GHermeneus/include/GHermeneus/Instruction.h

```diff
@@ -0,0 +1,95 @@
+//
+// Created by Jelle Spijker on 7/31/20.
+//
+
+#ifndef GCODEHERMENEUS_INSTRUCTION_H
+#define GCODEHERMENEUS_INSTRUCTION_H
+
+#include <string_view>
+#include <utility>
+#include <vector>
+
+#include "GCode.h"
+
+namespace GHermeneus
+{
+    /*!
+     * An dialect interpreted GCode line which consist of a corresponding line number, command key (e.q. G0, M104) and
+     * a vector of Parameters of type double.
+     *
+     * @brief An interpreted GCode function
+     * @tparam SSV_T the State Space Vector type
+     * @tparam T the magnitude type
+     */
+    template<typename SSV_T, typename T>
+    struct Instruction
+    {
+        Instruction() : line_no(-1), cmd("") {}; // Todo figure out why I need this constructor
+
+        Instruction(const size_t& line_no, const std::string_view& cmd, Parameters<T> params) :
+                line_no(line_no),
+                cmd(cmd),
+                params(std::move(params))
+        {};
+
+        Instruction(Instruction<SSV_T, T>&& instruction) noexcept:
+                line_no(instruction.line_no),
+                cmd(instruction.cmd),
+                params(std::move(instruction.params))
+        {};
+
+        Instruction(const Instruction<SSV_T, T>& instruction) noexcept:
+                line_no(instruction.line_no),
+                cmd(instruction.cmd),
+                params(instruction.params)
+        {};
+
+        const size_t line_no; //<! The line number
+        const std::string_view cmd; //<! The command key
+        Parameters<T> params; //<! A vector of extracted parameters
+
+        Instruction<SSV_T, T>& operator=(const Instruction<SSV_T, T>& rhs)
+        {
+            return *this;
+        }
+
+        bool operator<(const Instruction<SSV_T, T>& rhs) const
+        {
+            return line_no < rhs.line_no;
+        };
+
+        bool operator>(const Instruction<SSV_T, T>& rhs) const
+        {
+            return rhs < *this;
+        };
+
+        bool operator<=(const Instruction<SSV_T, T>& rhs) const
+        {
+            return !(rhs < *this);
+        };
+
+        bool operator>=(const Instruction<SSV_T, T>& rhs) const
+        {
+            return !(*this < rhs);
+        };
+
+        /*!
+         * Returns the delta State Space Vector according to a dialect transform.
+         * The previous extracted cmd key is used to look up the corresponding function. To which the previous extracted
+         * parameters and their value are passed. It will return the delta State Space Vector which can be added to the
+         * n - 1  State Space Vector or subtracted from the n + 1 State Space Vector.
+         *
+         * Todo: Keep in mind that the delta State Space Vector might also be an absolute State Space Vector
```

Comment:
Perhaps find some way to distinguish between the two? If only for debuggings' sake.

Reply:
Again. Whe are of one mind. In all probability the underlying Eigen matrix Will stay the same for both but it should be clear for debugging. I also still need to think about a state space vector were only a sub set of magnitudes is absolute and others are relative. Such as an absolute extrusion `e` and relative movements `x, y, z`.
Maybe I should create my own 'primitive type' for that which will replace the double or int in the Eigen Matrix.
On a side note I also wanted to use the boost unit lib or the https://github.com/mpusz/units lib for the primitive type. I prefer all my engineering programs to be unit safe.